### ISSUE TITLE
Issues when trying to build a 6-track GDI with buildgdi 2.0

### ISSUE THREAD

**Issue by IlDucci** [Sappharad/GDIbuilder#8]
Greetings. I've been trying a couple of the 2.0 betas of the buildgdi tool and also the still-hot final version to modify the PAL version of Resident Evil 2, and I am not able to get a working image. I'm working on W10.

This is my command line example:
```
_Tools\buildgdi.exe -rebuild -data data1 -ip IP.1.EDIT.BIN -V RESIDENTEVIL2 -output GDI1\ -gdi "(imagepath).gdi"
```

The resulting GDI fails to load properly even on GD-ROM explorer, where while it will detect all the tracks, it will only display information of track 01, which is left untouched by the program.

I have also found a minor, dumb mistake when trying to get a proper command, as I had the -cdda tracks invoked as well and that caused an error of "file already in use", but the tool no longer needs the -cdda option when going for -rebuild. You might want to have a graceful error handling there.

**Comment by Sappharad** [comment]
I don't know if I'll have time to look into this for a few days, but the CDDA argument being accepted for rebuild is intentional. I guess it's one of the things I either forgot to document or randomly mentioned it on discord and nowhere else.

The intent is that you're allowed to change the CDDA for your modified disc if you want to. If you specify -cdda, then it's going to copy the files that you specify into the new disc's folder and those will be the CDDA tracks for you new disc. If you don't specify CDDA, it reads the list from the existing GDI and copies those. What is probably missing is a check to prevent you from specifying CDDA tracks that are in your destination folder, because you can't overwrite the thing that you're reading from.

I might have never tested rebuild with CDDA, I will confirm when I have time.

**Comment by Sappharad** [comment]
I've done some additional testing and I'm not running into any problems. I successfully built a patched copy of a 19 track GDI without any issues, and both GD-ROM Explorer and GDIBuilder can read the disc and it also boots and runs fine. I'm wondering if this is a matter of you not understanding how to use this feature, and me not blocking some kind of incorrect use case. I already block some, but I've clearly missed something.

As it's described in the main menu of the GUI version, the rebuild feature is intended to be used to make a copy of a disc with some files replaced or added.

At the bare minimum you provide an original unmodified GDI as input, a data folder of files, and the output folder. The output folder should not contain any of the input, because we need to read that input to create the new disc. If the data folder contains 0GDTEX.PVR, then the original disc will be copied and only 0GDTEX.PVR will be modified. If the data folder contains a subfolder called AUDIO and a file MUSIC.ADX, and that same folder and file exists on the original disc, then it will be modified on your new disc. If that folder and file does not exist on the old disc, it will be added to the new disc while retaining all other files on the original disc.

If your data folder contains everything on a disc, then you shouldn't be using rebuild, you should be building a new disc. 

The GUI only offers the most common options, (input, folder for files, and output folder) but the command line version allows you to extra things like replace the IP.BIN or CDDA tracks with different ones. If those arguments are not specified, the new disc uses the files from the old disc.

Nothing seems off about your usage example, assuming data1 is a folder with only files you want to change, but is it possible you're putting more files into the folder than will fit on a disc? The program isn't supposed to allow that and should throw an error.

Please let me know if you've had any luck after your original post or if you're able to narrow down at all what doesn't work for you. Perhaps try a smaller example where you just replace a single file and go from there to figure out what causes it to stop working?

Thanks in advance if there's anything you can do to help me understand why it's not working for you.

**Comment by Sappharad** [comment]
I believe this was the same issue that I fixed in v2.1
The file table was in the wrong place and being overwritten by files when a disc had over 492MB of files on it.
